ICSharpCode.Decompiler.Tests/TestCases/Pretty/Structs.cs

```diff
@@ -51,4 +51,28 @@ public StructWithDefaultCtor()
 		}
 	}
 #endif
-}
\ No newline at end of file
+
+#if CS110
+	public struct StructWithRequiredMembers
+	{
+		public required string FirstName;
+		public required string LastName { get; set; }
+	}
+#endif
+}
+
+namespace System.Runtime.CompilerServices
+{
+	[AttributeUsage(AttributeTargets.All, AllowMultiple = true, Inherited = false)]
+	internal sealed class CompilerFeatureRequiredAttribute : Attribute
```

Comment:
I think we should add a transform that removes these, if the target framework is net7.0.

Also note that https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4AEBMBGAsAKHQGYACLEgYRIG8SCSGzSAnAUwEcBXKNhE2AC4kAgjRIBzVgIDcJAM5TZAX3qNiJNlx6s+gkgDFpBJUA== fully qualifies the `RefSafetyRules` module attribute. This could be because it is defined in two places, i.e., the preview framework used by sharplab as well as the local definition emitted by the compiler.

Reply:
These attributes were necessary for the test case to compile on older dotnet versions.
I think we should output the attribute if it is present in the assembly. It doesn't get embedded automatically (unlike some other attributes).